File: src/platform/mobile-bff/src/services/order.js
// TODO: Replace stub methods with @grpc/grpc-js client calls once proto/commerce/ is compiled.
export class OrderService {
  constructor (addr) { this.addr = addr }

  async listOrders (_userId, _page, _pageSize) { throw notImplemented() }
  async getOrder (_orderId, _userId)           { throw notImplemented() }
  async placeOrder (_userId, _req)             { throw notImplemented() }
}

function notImplemented () {
  const err = new Error('gRPC client not yet implemented — pending proto compilation')
  err.code = 'NOT_IMPLEMENTED'
  return err
}

File: src/platform/web-bff/internal/service/order.go
package service

import (
	"context"

	"github.com/shopos/web-bff/internal/domain"
)

type OrderService interface {
	ListOrders(ctx context.Context, userID string, page, pageSize int) ([]*domain.Order, error)
	GetOrder(ctx context.Context, orderID, userID string) (*domain.Order, error)
	PlaceOrder(ctx context.Context, userID string, req *domain.PlaceOrderRequest) (*domain.Order, error)
}

// GRPCOrderService connects to commerce/order-service (port 50082).
// TODO: Replace stub bodies with generated proto client calls once proto/commerce/ is compiled.
type GRPCOrderService struct{ addr string }

func NewGRPCOrderService(addr string) OrderService {
	return &GRPCOrderService{addr: addr}
}

func (s *GRPCOrderService) ListOrders(_ context.Context, _ string, _, _ int) ([]*domain.Order, error) {
	return nil, ErrNotImplemented
}

func (s *GRPCOrderService) GetOrder(_ context.Context, _, _ string) (*domain.Order, error) {
	return nil, ErrNotImplemented
}

func (s *GRPCOrderService) PlaceOrder(_ context.Context, _ string, _ *domain.PlaceOrderRequest) (*domain.Order, error) {
	return nil, ErrNotImplemented
}
